src/async_asset.rs

```diff
@@ -29,6 +28,125 @@ impl AsyncAssetTaskExt for TaskContext {
     }
 }
 
+/// Because we can't implement [PartialEq] on a foreign type, create our own trait that mirrors the interface
+trait PartialEquality {
+    fn eq(&self, other: &Self) -> bool;
+}
+
+impl PartialEquality for AssetLoadError {
```

Comment:
This is the main issue that blocked me finishing this myself. I'm happy with this solution for now since it's a solution, but I'll want to revisit this to convince myself there's not a better way. Leaving this comment mostly for myself for reference.

Reply:
With some of the inner error types, their fields are private, but they get printed out in their `Display` implementation, so I just compare their string representations. I guess you could do that for the entire AssetLoadError (though you might want to add a test to check that all the error types have sufficiently disambiguating string representatiions)